README.md

```diff
@@ -15,7 +15,33 @@ The spec allows you to define:
 * Desired OpenShift version
 * Desires operators versions
 
-A set of example CRs can be found on the *samples* folder.
+A set of example CRs can be found on the **samples** folder.
+
+## How it works
+
+The ClusterGroupUpgrade controller is designed to perform an upgrade to a group of clusters
+in a given period of time, which by default is set to 4h.
+In order to model the various phases of an upgrade, the ClusterGroupUpgrade CR can be in the
+following states:
+
+* **UpgradeNotStarted**
+  * In this state, the **ClusterGroupUpgrade** CR has just been created and the *remediationAction* field is set to *inform*
+  * The controller will build a remediation plan based on the *clusters* list and with *remediationStrategy* fields. For example, if the *clusters* list is made of 6 clusters and the *remediationStrategy* has *maxConcurrency* set to 2 the controller will create a remediation plan with 3 batches: the first will contain the first two clusters, the second the third and the fourth and the third batch the fifth and the sixth. In another example, if the *clusters* list is made of the same 6 clusters and the *remediationStrategy* has *maxConcurrency* set to 2 but *canaries* list contains the second cluster from the *clusters* list then the controller will build a remediation plan of 4 batches: the first will contain the second cluster, the second batch will contain the first and the third, the third batch will contain the fourth and the fifth and the fourth batch will contain the sixth cluster.
```

Comment:
"the second the third and the fourth and the third batch the fifth and the sixth" - This phrasing is a bit confusing. Maybe we could say "the second batch with contain clusters 3 & 4 and the 3rd batch clusters 5 & 6" or something similar.